includes/admin/marketplace-suggestions/class-wc-marketplace-suggestions.php

```diff
@@ -0,0 +1,161 @@
+				$dismissed_suggestions = array();
+			}
+		}
+
+		return $dismissed_suggestions;
+	}
+
+	/**
+	 * POST handler for adding a dismissed suggestion.
+	 */
+	public static function post_add_dismissed_suggestion_handler() {
+		if ( ! check_ajax_referer( 'add_dismissed_marketplace_suggestion' ) ) {
+			wp_die();
+		}
+
+		$post_data       = wp_unslash( $_POST );
+		$suggestion_slug = sanitize_text_field( $post_data['slug'] );
+		if ( ! $suggestion_slug ) {
+			wp_die();
+		}
+
+		$dismissed_suggestions = self::get_dismissed_suggestions();
+
+		if ( in_array( $suggestion_slug, $dismissed_suggestions, true ) ) {
+			wp_die();
+		}
+
+		$dismissed_suggestions[] = $suggestion_slug;
+		update_user_meta(
+			get_current_user_id(),
+			'wc_marketplace_suggestions_dismissed_suggestions',
+			$dismissed_suggestions
+		);
+
+		wp_die();
+	}
+
+	/**
+	 * Render suggestions containers in products list empty state.
+	 */
+	public static function render_products_list_empty_state() {
+		self::render_suggestions_container( 'products-list-empty-header' );
+		self::render_suggestions_container( 'products-list-empty-body' );
+		self::render_suggestions_container( 'products-list-empty-footer' );
+	}
+
+	/**
+	 * Render a suggestions container element, with the specified context.
+	 *
+	 * @param string $context Suggestion context name (rendered as a css class).
+	 */
+	public static function render_suggestions_container( $context ) {
+		include dirname( __FILE__ ) . '/views/container.php';
+	}
+
+	/**
+	 * Should suggestions be displayed?
+	 *
+	 * @return bool
+	 */
+	public static function allow_suggestions() {
+		// We currently only support English suggestions.
+		$locale             = get_locale();
+		$suggestion_locales = array(
+			'en_AU',
+			'en_CA',
+			'en_GB',
+			'en_NZ',
+			'en_US',
+			'en_ZA',
+		);
+		if ( ! in_array( $locale, $suggestion_locales, true ) ) {
+			return false;
+		}
+
+		// Suggestions are only displayed if user can install plugins.
+		if ( ! current_user_can( 'install_plugins' ) ) {
+			return false;
+		}
+
+		// User can disabled all suggestions via filter.
+		return apply_filters( 'woocommerce_allow_marketplace_suggestions', true );
+	}
+
+	/**
+	 * Pull suggestion data from remote endpoint & cache in a transient.
+	 *
+	 * @return array of json API data
+	 */
+	public static function get_suggestions_api_data() {
+		$suggestion_data = get_transient( 'wc_marketplace_suggestions' );
+		if ( false !== $suggestion_data ) {
+			return $suggestion_data;
+		}
+
+		$suggestion_data_url = 'https://d3t0oesq8995hv.cloudfront.net/add-ons/marketplace-suggestions.json';
+		$raw_suggestions     = wp_remote_get(
+			$suggestion_data_url,
+			array( 'user-agent' => 'WooCommerce Marketplace Suggestions' )
+		);
+
+		// Parse the data to check for any errors.
+		// If it's valid, store structure in transient.
+		if ( ! is_wp_error( $raw_suggestions ) ) {
+			$suggestions = json_decode( wp_remote_retrieve_body( $raw_suggestions ) );
```

Comment:
Does json_decode need second argument to force assoc array vs object here? Not sure, just asking.

Reply:
Correct – [the default is `bool $assoc = FALSE`](https://secure.php.net/manual/en/function.json-decode.php). This is fixed in the `WC_Queue` updater:

https://github.com/woocommerce/woocommerce/pull/22857/files#diff-7cb8cd3240aef8097b64fc2183b79fb1R66